freqtrade/exchange/bybit.py

```diff
@@ -233,3 +234,25 @@ def fetch_order(self, order_id: str, pair: str, params: Dict = {}) -> Dict:
             # Canceled orders will have "remaining=0" on bybit.
             order['remaining'] = None
         return order
+
+    def cancel_order_with_result(self, order_id: str, pair: str, amount: float) -> Dict:
+        order = super().cancel_order_with_result(order_id, pair, amount)
+
+        # Currently bybit does not update status of the cancelled order on time.
+        # Even though endpoint returns proper response and code 200, the status stays open.
+        # This re-fetches the order giving bybit enough time to update the status.
+        if order.get("status") == "open" and order.get("filled") == 0.0:
+            try:
+                order = self.fetch_order(order_id, pair)
+            except InvalidOrderException:
+                logger.warning(f"Could not fetch cancelled order {order_id}.")
+                order = {
```

Comment:
I don't think you'll need this here.

I understand it's a copy/paste from the upstream method - but it really isn't necessary (it's a workaround for exchanges that do a "/delete" on canceled orders, and don't remember them seconds after they're canceled).

if the order disappears (which would mean bybit forgets about the order - which bybit doesn't do, at least for several days) - then you'll get a "canceled" response from the parent method already - never actually hitting this branch.